### ISSUE TITLE
"Align" rule doesn't consider access modifiers

### ISSUE THREAD

**Issue by newsash** [palantir/tslint#4537]
### Bug Report

- __TSLint version__: ~5.12.1
- __TypeScript version__: ~3.2.4
- __Running TSLint via__: CLI ("ng lint") / VSCode

#### TypeScript code being linted

```ts
constructor(public id: IIdentity,
                   dateTimeStart: Date | null = null,
                   dateTimeEnd: Date | null = null) {
```

with `tslint.json` configuration:

```json
{
  "extends": "tslint:recommended",
  "rulesDirectory": [
    "codelyzer"
  ],
  "rules": {
  },
}
```

#### Actual behavior
Suggests:
```ts
constructor(public id: IIdentity,
            dateTimeStart: Date | null = null,
            dateTimeEnd: Date | null = null) {
```
#### Expected behavior
Suggests:
```ts
constructor(public id: IIdentity,
                   dateTimeStart: Date | null = null,
                   dateTimeEnd: Date | null = null) {
```

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
